component/otelcol/exporter/prometheus/internal/convert/convert.go

```diff
@@ -524,10 +563,44 @@ func (conv *Converter) consumeHistogram(app storage.Appender, memResource *memor
 			if err := writeSeries(app, infBucket, dp, infBucketVal); err != nil {
 				level.Error(conv.log).Log("msg", "failed to write histogram bucket sample", "bucket", bucketLabel.Value, "err", err)
 			}
+
+			// Add remaining exemplars.
+			for j := exemplarsIndex; j < len(exemplars); j++ {
+				if err := conv.writeExemplar(app, infBucket, exemplars[j]); err != nil {
+					level.Error(conv.log).Log("msg", "failed to add exemplar to bucket", bucketLabel.Value, "err", err)
+				}
+			}
 		}
 	}
 }
 
+// Convert Otel Exemplar to Prometheus Exemplar.
+func (conv *Converter) convertExemplar(otelExemplar pmetric.Exemplar, ts time.Time) exemplar.Exemplar {
+	exemplarLabels := make(labels.Labels, 0)
+
+	if traceID := otelExemplar.TraceID(); !traceID.IsEmpty() {
+		exemplarLabels = append(exemplarLabels, labels.Label{Name: "trace_id", Value: hex.EncodeToString(traceID[:])})
```

Comment:
I don't think you need to create a new slice here with `[:]` cause encoding is read-only? LMK if I'm missing something.

Reply:
traceID is an array (type TraceID [16]byte) and EncodeToString is expecting a slice so I believe that we need to convert it into a slice this way (cant pass traceID directly

Reply:
ah right, it has a fixed size `[16]` 👍